Source/WebCore/platform/DateTimeChooser.h

```diff
@@ -31,27 +31,19 @@
 #pragma once
 
 #if ENABLE(DATE_AND_TIME_INPUT_TYPES)
-#include <wtf/TZoneMallocInlines.h>
-#include <wtf/WeakPtr.h>
-
-namespace WebCore {
-class DateTimeChooser;
-}
-
-namespace WTF {
-template<typename T> struct IsDeprecatedWeakRefSmartPointerException;
-template<> struct IsDeprecatedWeakRefSmartPointerException<WebCore::DateTimeChooser> : std::true_type { };
-}
+#include <wtf/CanMakeWeakPtr.h>
 
 namespace WebCore {
 
 struct DateTimeChooserParameters;
 
 class DateTimeChooser : public CanMakeWeakPtr<DateTimeChooser> {
-    WTF_MAKE_TZONE_ALLOCATED_INLINE(DateTimeChooser);
```

Comment:
Why did you drop this?

Reply:
When I made `WebDateTimeChooser` ref-counted, it gave me an error saying that it had two `operator new`'s (one from `RefCounted` and one from `WTF_MAKE_TZONE_ALLOCATED_INLINE(DateTimeChooser)`. This was fixable by adding `WTF_MAKE_TZONE_ALLOCATED_INLINE(WebDateTimeChooser)` to `WebDateTimeChooser`. 

Then I thought that if no other class inherits from `DateTimeChooser` and `WebDateTimeChooser` already has `WTF_MAKE_TZONE_ALLOCATED_INLINE`, then there's no reason for `DateTimeChooser` to have it. Also no one instantiates an object of `DateTimeChooser` type--so it felt like there was no reason to specify how it's allocated.

This is model that `SpeechSynthesisClient` follows as well.

Reply:
But honestly, I don't quite understand how the TZONE_ALLOCATION thing works--so maybe this is the wrong approach?

Reply:
Here is an example of code using TZONE and subclassing RefCounted in our code base:
```
class ViewTransitionTypeSet : public RefCounted<ViewTransitionTypeSet> {
    WTF_MAKE_TZONE_OR_ISO_ALLOCATED(ViewTransitionTypeSet);
```

Maybe you can try to mimic that we do for this class to try and get the build working. I don't know much about TZONE either bug I know we want more TZONE, not less :D